src/TestJourney.elm

```diff
@@ -1091,6 +1091,136 @@ stepSimulateEvent description event finder =
         )
 
 
+verticalBar : String -> String -> String -> String
+verticalBar comparison expected actual =
+    [ actual
+    , "╵"
+    , "│ " ++ comparison
+    , "╷"
+    , expected
+    ]
+        |> String.join "\n"
+
+
+toStringLists : List String -> String
+toStringLists =
+    String.join ", "
+
+
+listDiffToString :
+    Int
+    -> String
+    -> { expected : List String, actual : List String }
+    -> { originalExpected : List String, originalActual : List String }
+    -> String
+listDiffToString index description { expected, actual } originals =
+    case ( expected, actual ) of
+        ( [], [] ) ->
+            [ "Two lists were unequal previously, yet ended up equal later."
+            , "This should never happen!"
+            , "Please report this bug to https://github.com/elm-community/elm-test/issues - and include these lists: "
+            , "\n"
+            , toStringLists originals.originalExpected
+            , "\n"
+            , toStringLists originals.originalActual
+            ]
+                |> String.join ""
+
+        ( _ :: _, [] ) ->
+            verticalBar (description ++ " was shorter than")
+                (toStringLists originals.originalExpected)
+                (toStringLists originals.originalActual)
+
+        ( [], _ :: _ ) ->
+            verticalBar (description ++ " was longer than")
+                (toStringLists originals.originalExpected)
+                (toStringLists originals.originalActual)
+
+        ( firstExpected :: restExpected, firstActual :: restActual ) ->
+            if firstExpected == firstActual then
+                -- They're still the same so far; keep going.
+                listDiffToString (index + 1)
+                    description
+                    { expected = restExpected
+                    , actual = restActual
+                    }
+                    originals
+
+            else
+                -- We found elements that differ; fail!
+                String.join ""
+                    [ verticalBar description
+                        (toStringLists originals.originalExpected)
+                        (toStringLists originals.originalActual)
+                    , "\n\nThe first diff is at index "
+                    , String.fromInt index
+                    , ": it was `"
+                    , firstActual
+                    , "`, but `"
+                    , firstExpected
+                    , "` was expected."
+                    ]
+
+formatFailure : String -> Test.Runner.Failure.Reason -> String
```

Comment:
Interesting!  I guess we had to inline this, because it was in a different namespace in two elm test versions?

I'm good w/ updating the `elm-explorations/test` dependency to `2.0.1 <= v < 3.0.0` if it removes the needs for this.

Reply:
This was to handle the removal of `format` from the exposed `test` API.  It was moved to a test only package in version `2.0.0`.  So upgrading dependency requirements will not fix... at least to my understanding it wont.

Heres the change log alert:
https://github.com/elm-explorations/test/blob/2.0.1/CHANGELOG.md#5-testrunnerfailureformat-removal

As I said, the `format` function still exists in the code base and is just unexposed. Looks like they made changes to it also but I wasn't sure why so kept to the old version of the code.

Heres the old/new format:
https://github.com/elm-explorations/test/blob/2.0.1/tests/src/Runner/String/Format.elm

I altered the PR moving our `format` to a separate file to avoid clutter.

Any suggestions you have would of course be appreciated.  I wasn't thrilled with the copy but not sure how to avoid.

Reply:
Thanks @jalandis !  I totally missed that it was moved into the tests.  I can't think of a better way w/o doing a good amount of rethinking of this library.  Thanks for splitting it out!